components/src/core/components/Button/Icon.vue

```diff
@@ -77,6 +77,10 @@ export default defineComponent({
       type: Boolean,
       default: false,
     },
+    noFocus: {
```

Comment:
Why do we need this new prop to handle the icon highlighting after clicking on it. I feel it's an overkill. @Super-Chama any comments on this?

Reply:
Can't say for sure without knowing the use case of this. why do we need to disable the focus here? if it's default behaviour we can remove it outright without prop

Reply:
This is added due to the reopen scenario of RMP-594

https://orangehrmenterprise.atlassian.net/browse/RMP-594?focusedCommentId=15545
After clicking on the reset icon, still, the icon is in highlighted state. Discussed with mike and it is not the expected behavior.

Reply:
Had a discussion with @gayan-orangehrm I feel like we can easily remove focus from element rather manually setting styles. can modify the existing `onClick()` as follows

```
onClick(e: Event) {
  e.preventDefault();
  (e.target as HTMLElement).blur();
  this.$emit('click', e);
}
```

Reply:
Agreed to keep the current behaviour where the focus remains at the reset button and the colour difference is okay to have till clicking outside.